.github/workflows/docstrings.yml

```diff
@@ -20,10 +20,9 @@ jobs:
     - name: Install Dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install pytest
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        python -m pip install meson-python meson ninja pybind11 versioningit pytest
```

Comment:
As these are already in build requirements, we can drop the explicit install here. They will be picked up by `pip install` below

Reply:
For an editable install, they [appear](https://mesonbuild.com/meson-python/how-to-guides/editable-installs.html#build-dependencies) to be needed by meson... Are you saying use `pip install --no-build-isolation -e "[build]"`?

Reply:
Hmm I see. Sadly there is [not a more ergonomic way to do this with `pip`]( https://github.com/pypa/pip/issues/12117 )

Agree the best approach unfortunately is to reproduce the list here

Reply:
ya, pretty clunky.

thanks for your eyes on this!